src/flamegraph/merge.rs

```diff
@@ -148,8 +172,29 @@ where
             iter::once("").chain(last.split(';')),
             None,
             time,
+            delta,
         );
     }
 
-    (frames, time, ignored)
+    (frames, time, ignored, delta_max)
+}
+
+// Parse and remove the number of samples from the end of a line.
+fn parse_nsamples(line: &mut &str) -> Option<usize> {
+    if let Some(samplesi) = line.rfind(' ') {
```

Comment:
You can take advantage of the fact that `Option` implements [`Try`](https://doc.rust-lang.org/std/ops/trait.Try.html) here! Try:
```rust
let samplesi = line.rfind(' ')?;
```
You can probably use that to simplify the code further down too!